pmd-apex/src/main/resources/category/apex/codestyle.xml

```diff
@@ -149,6 +149,35 @@ public class Foo {
         </example>
     </rule>
 
+    <rule name="OneDeclarationPerStatement"
+          since="6.7.0"
+          message="Use one line for each declaration, it enhances code readability."
+          class="net.sourceforge.pmd.lang.apex.rule.ApexXPathRule"
+          externalInfoUrl="${pmd.website.baseurl}/pmd_rules_apex_codestyle.html#onedeclarationperstatement">
+        <description>
+Apex allows the use of several variables declaration of the same type on one line. However, it
+can lead to quite messy code. This rule looks for several declarations on the same line.
+        </description>
+        <priority>1</priority>
+        <properties>
+            <property name="xpath">
+                <value>
+<![CDATA[
+//VariableDeclarationStatements[VariableDeclaration[2]]|//FieldDeclarationStatements[FieldDeclaration[2]]
```

Comment:
I'd find it more readable to use the count function, e.g. `FieldDeclarationStatements[count(FieldDeclaration) > 1]`, and to  put the two alternatives on separate lines